### PR TITLE
Support for environmental groups install / remove / upgrade

### PR DESCRIPTION
**Pull request by m-blaha**
With this patch all `dnf5 group` subcommands can also handle environmental groups.

Requires: https://github.com/rpm-software-management/dnf5/pull/554
Resolves: https://github.com/rpm-software-management/dnf5/issues/367

### CODE REVIEW

File: libdnf/base/goal.cpp [hunk 1]
```diff
@@ -1638,6 +1699,163 @@ void Goal::Impl::add_group_upgrade_to_goal(
...
+    auto & cfg_main = base->get_config();
+    bool with_optional = any(settings.resolve_group_package_types(cfg_main) & libdnf::comps::PackageType::OPTIONAL);
+    auto group_settings = libdnf::GoalJobSettings(settings);
+    group_settings.group_search_environments = false;
+    std::vector<GroupSpec> env_group_specs;
+    for (auto environment : environment_query) {
+        rpm_goal.add_environment(environment, transaction::TransactionItemAction::INSTALL, with_optional);
+        for (const auto & grp_id : environment.get_groups()) {
+            env_group_specs.emplace_back(
+                GoalAction::INSTALL_BY_GROUP, transaction::TransactionItemReason::DEPENDENCY, grp_id, group_settings);
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply libdnf/base/goal.cpp:1724]: I didn't see much advantage of adding a new action for environments. I can rename the current action to something like `GoalAction::INSTALL_BY_COMPS`, though.

File: libdnf/base/goal.cpp [hunk 2]
```diff
@@ -1773,6 +1779,83 @@ void Goal::Impl::add_environment_remove_to_goal(
...
+                    env_group_specs.emplace_back(
+                        GoalAction::UPGRADE, transaction::TransactionItemReason::DEPENDENCY, grp, group_settings);
+                } catch (const system::StateNotFoundError &) {
+                    continue;
+                }
+            } else {
+                // newly added group to environment definition, install it.
+                env_group_specs.emplace_back(
+                    GoalAction::INSTALL_BY_GROUP, transaction::TransactionItemReason::DEPENDENCY, grp, group_settings);
+            }
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply libdnf/base/goal.cpp:1853]: Oh, yup. It somehow dropped off the final version of the PR. I'll fix it.

File: libdnf/base/goal.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kontura) [reply libdnf/base/goal.cpp:1724]: I think that would be a good change, since we can still change the API. :+1:

**Review by kontura — APPROVED** [review]
Thanks!

**Pull request merged**